packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts

```diff
@@ -118,16 +122,22 @@ export class UPlotConfigBuilder {
 
     config.tzDate = this.tzDate;
 
-    // When bands exist, only keep fill when defined
-    if (this.bands?.length) {
+    if (this.isStacking) {
+      // Let uPlot handle bands and fills
       config.bands = this.bands;
-      const keepFill = new Set<number>();
-      for (const b of config.bands) {
-        keepFill.add(b.series[0]);
-      }
-      for (let i = 1; i < config.series.length; i++) {
-        if (!keepFill.has(i)) {
-          config.series[i].fill = undefined;
+    } else {
+      // When fillBelowTo option enabled, handle series bands fill manually
+      if (this.bands?.length) {
+        config.bands = this.bands;
+        const keepFill = new Set<number>();
+        for (const b of config.bands) {
+          keepFill.add(b.series[0]);
+        }
+
+        for (let i = 1; i < config.series.length; i++) {
+          if (!keepFill.has(i)) {
+            config.series[i].fill = undefined;
+          }
```

Comment:
This makes the stacking and fillBelowTo options mutually exclusive. We talked this through with @leeoniya and this makes most sense as it's most probably super uncommon to have these two combined.